tests/test_quick_tests.py

```diff
@@ -754,3 +755,61 @@ def test_random_generation_without_range_values(self, columnSpecOptions):
     def test_version_info(self):
         # test access to version info without explicit import
         print("Data generator version", dg.__version__)
+
+    def test_multi_column_generation(self):
+        column_specs = [
+            {"colName": "col1", "colType": "int", "minValue": 0, "maxValue": 100},
+            {"colName": "col2", "colType": "float", "minValue": 0.0, "maxValue": 100.0},
+            {"colName": "col3", "colType": "string", "values": ["a", "b", "c"], "random": True}
+        ]
+        df_from_dicts = dg.DataGenerator(rows=100, partitions=1).withColumns(column_specs).build()
+        assert df_from_dicts.columns == ["col1", "col2", "col3"]
+
+    def test_generation_from_dictionary(self):
+        dg_spec = {
+            "name": "test_data_generator",
+            "rows": 1000,
+            "partitions": 10,
+            "randomSeedMethod": "fixed",
+            "randomSeed": 42,
+            "random": True
+        }
+        gen_from_dict = DataGenerator.fromDict(dg_spec)
+        assert gen_from_dict.name == dg_spec.get("name")
+        assert gen_from_dict.rowCount == dg_spec.get("rows")
+        assert gen_from_dict.partitions == dg_spec.get("partitions")
+        assert gen_from_dict.random == dg_spec.get("random")
+        assert gen_from_dict.randomSeed == dg_spec.get("randomSeed")
+
+    def test_generation_from_file(self):
+        path = "tests/files/test_generator_spec.json"
```

Comment:
If we use string based APIs, they'll be more general - also you can simply define the definitions as multi-line strings rather than requiring separate data files